custom_metrics/fugu-apis.js

```diff
@@ -0,0 +1,213 @@
+  },
+  "WebOTP": {
+    regEx: /navigator\.credentials\.get\(/g,
+    where: "JavaScript",
+  },
+  "File System Access": {
+    regEx: /showOpenFilePicker\(/g,
+    where: "JavaScript",
+  },
+  "Pointer Lock (unadjustedMovement)": {
+    regEx: /unadjustedMovement\:/g,
+    where: "JavaScript",
+  },
+  "WebHID": {
+    regEx: /navigator\.hid\.requestDevice\(/g,
+    where: "JavaScript",
+  },
+  "WebSerial": {
+    regEx: /navigator\.serial\.requestPort\(/g,
+    where: "JavaScript",
+  },
+  "WebNFC": {
+    regEx: /new NDEFReader\(/g,
+    where: "JavaScript",
+  },
+  "Run On Login": {
+    regEx: /navigator\.runOnOsLogin\.set\(/g,
+    where: "JavaScript",
+  },
+  "WebCodecs": {
+    regEx: /new MediaStreamTrackProcessor\(/g,
+    where: "JavaScript",
+  },
+  "Digital Goods": {
+    regEx: /getDigitalGoodsService\(/g,
+    where: "JavaScript",
+  },
+  "Idle Detection": {
+    regEx: /new IdleDetector\(/g,
+    where: "JavaScript",
+  },
+  "Storage Foundation": {
+    regEx: /storageFoundation\.open\(/g,
+    where: "JavaScript",
+  },
+  "Handwriting Recognition": {
+    regEx: /navigator\.queryHandwritingRecognizerSupport\(/g,
+    where: "JavaScript",
+  },
+  "Compute Pressure": {
+    regEx: /new ComputePressureObserver\(/g,
+    where: "JavaScript",
+  },
+  "Ambient Light Sensor": {
+    regEx: /new AmbientLightSensor\(\)/g,
+    where: "JavaScript",
+  },
+  "File Handling": {
+    regEx: /launchQueue\.setConsumer\(/g,
+    where: "JavaScript",
+  },
+  "Notification Triggers": {
+    regEx: /showTrigger\:/g,
+    where: "JavaScript",
+  },
+  "Local Font Access": {
+    regEx: /navigator\.fonts\.query\(\);/g,
+    where: "JavaScript",
+  },
+  "Multi-Screen Window Placement": {
+    regEx: /getScreens\(\)/g,
+    where: "JavaScript",
+  },
+  "WebSocketStream": {
+    regEx: /new WebSocketStream\(/g,
+    where: "JavaScript",
+  },
+  "WebTransport": {
+    regEx: /new WebTransport\(/g,
+    where: "JavaScript",
+  },
+  "Web Share Target": {
+    regEx: /"share_target"\:/g,
+    where: "Web App Manifest",
+  },
+  "Web Share Target (Files)": {
+    regEx: /"enctype"\: "multipart\/form\-data"/g,
+    where: "Web App Manifest",
+  },
+  "Shortcuts": {
+    regEx: /"shortcuts"\:/g,
+    where: "Web App Manifest",
+  },
+  "Declarative Link Capturing": {
+    regEx: /"capture_links"\:/g,
+    where: "Web App Manifest",
+  },
+  "Tabbed Application Mode": {
+    regEx: /"tabbed"/g,
+    where: "Web App Manifest",
+  },
+  "Window Controls Overlay": {
+    regEx: /"window\-controls\-overlay"/g,
+    where: "Web App Manifest",
+  },
+  "URL Handlers": {
+    regEx: /"url_handlers"/g,
+    where: "Web App Manifest",
+  },
+  "Protocol Handlers": {
+    regEx: /"protocol_handlers"/g,
+    where: "Web App Manifest",
+  },
+};
+
+// To make sure we don't match on, e.g., blog posts that contain the patterns,
+// make sure that the file names fulfill certain conditions as a heuristic.
+const checkURLConditions = (where, url) => {
```

Comment:
Would it be helpful for minimizing false negatives to also check the MIME type?

Reply:
Let's do it. I'll be forgiving and accept anything that looks right enough.